syntax = "proto3";

package grpcserver;

service GRPCServer {
    rpc push_message(Message) returns (Void);
    rpc pop_message(Timeout) returns (Message);
    rpc push_browser_request(BrowserRequest) returns (Void);
    rpc pop_browser_request(Timeout) returns (BrowserRequest);
    rpc len_browser_request(Void) returns (QueueSize);
    rpc push_browser_response(BrowserResponse) returns (Void);
    rpc pop_browser_response(Timeout) returns (BrowserResponse);
}

message Void {}

message Message {
    bytes content = 1;
}

message BrowserRequest {
    bytes content = 1;
}

message BrowserResponse {
    bytes content = 1;
}

message QueueSize {
    uint32 size = 1;
}

message Timeout {
    double seconds = 1;
}
